tests/integration/test_route53resolver.py

```diff
@@ -0,0 +1,565 @@
+                ResolverEndpointId=create_resolver_endpoint_res["Id"]
+            )
+        )
+
+        self._wait_created_endpoint_is_listed_with_status(
+            route53resolver_client, request_id, "OPERATIONAL"
+        )
+        snapshot.match("create_resolver_endpoint_res", create_resolver_endpoint_res)
+
+    @pytest.mark.aws_validated
+    def test_route53resolver_bad_create_endpoint_security_groups(
+        self, ec2_client, route53resolver_client, snapshot
+    ):
+        request_id = short_uid()
+        resolver_endpoint_name = f"rs-{request_id}"
+        with pytest.raises(
+            route53resolver_client.exceptions.InvalidParameterException
+        ) as inavlid_param_request_res:
+            route53resolver_client.create_resolver_endpoint(
+                CreatorRequestId=request_id,
+                SecurityGroupIds=["test-invalid-sg-123"],
+                Direction="INBOUND",
+                Name=resolver_endpoint_name,
+                IpAddresses=self.get_create_resolver_endpoint_ip_address(ec2_client, "43"),
+            )
+        snapshot.match("inavlid_param_request_res", inavlid_param_request_res.value.response)
+
+    @pytest.mark.aws_validated
+    @pytest.mark.skip_snapshot_verify(paths=["$..SecurityGroupIds"])
+    def test_multiple_create_resolver_endpoint_with_same_req_id(
+        self, ec2_client, route53resolver_client, cleanups, snapshot
+    ):
+        request_id = short_uid()
+        resolver_endpoint_name = f"rs-{request_id}"
+        ip_addresses = self.get_create_resolver_endpoint_ip_address(ec2_client, "41")
+        security_groups_ids = self.get_security_group_ids(ec2_client)
+        result = route53resolver_client.create_resolver_endpoint(
+            CreatorRequestId=request_id,
+            SecurityGroupIds=security_groups_ids,
+            Direction="INBOUND",
+            Name=resolver_endpoint_name,
+            IpAddresses=ip_addresses,
+        )
+
+        create_resolver_endpoint_res = result.get("ResolverEndpoint")
+
+        cleanups.append(
+            lambda: route53resolver_client.delete_resolver_endpoint(
+                ResolverEndpointId=create_resolver_endpoint_res["Id"]
+            )
+        )
+
+        self._wait_created_endpoint_is_listed_with_status(
+            route53resolver_client, request_id, "OPERATIONAL"
+        )
+        snapshot.match("create_resolver_endpoint_res", create_resolver_endpoint_res)
+
+        with pytest.raises(
+            route53resolver_client.exceptions.ResourceExistsException
+        ) as res_exists_ex:
+            route53resolver_client.create_resolver_endpoint(
+                CreatorRequestId=request_id,
+                SecurityGroupIds=security_groups_ids,
+                Direction="INBOUND",
+                Name=resolver_endpoint_name,
+                IpAddresses=ip_addresses,
+            )
+
+        snapshot.match(
+            "res_exists_ex_error_code", res_exists_ex.value.response.get("Error").get("Code")
```

Comment:
```suggestion
            "res_exists_ex_error_code", res_exists_ex.value.response.get("Error", {}).get("Code")
```
Alternatively indexed access would be fine too since this shouldn't be None anyway